### PR TITLE
docs: update reference docs

### PR DESCRIPTION
**Pull request by desimone**
## Summary

Cleans up some language on the reference docs. Definitely could use another pass or two. 

**Checklist**:
- [x] updated docs
- [x] ready for review

### CODE REVIEW

File: docs/reference/readme.md [hunk 1]
```diff
@@ -18,11 +18,11 @@ Pomerium can be configured using a configuration file ([YAML]/[JSON]/[TOML]) or
 
 Using both [environmental variables] and config file keys is allowed and encouraged (for instance, secret keys are probably best set as environmental variables). However, if duplicate configuration keys are found, environment variables take precedence.
 
-Pomerium will automatically reload the configuration file if it is changed. At this time, only policy is re-configured when this reload occurs, but additional options may be added in the future. It is suggested that your policy is stored in a configuration file so that you can take advantage of this feature.
+Pomerium will automatically reload the configuration file if it is changed.
```
Comment (travisgroth) [inline docs/reference/readme.md:21]: This is more true than it used to be but not 100% coverage.  We might want to mark what is static (I believe it is only a subset of options).

File: docs/reference/readme.md [hunk 2]
```diff
@@ -18,11 +18,11 @@ Pomerium can be configured using a configuration file ([YAML]/[JSON]/[TOML]) or
...
 
 ## Shared Settings
 
-These are configuration variables shared by all services, in all service modes.
+These are configuration variables are shared by all services, in all service modes.
```
Comment (travisgroth) [inline docs/reference/readme.md:25]:
```suggestion
These configuration variables are shared by all services, in all service modes.
```

File: docs/reference/readme.md [hunk 3]
```diff
@@ -509,7 +508,7 @@ Log level sets the logging level for the pomerium proxy service. Only logs of th
 - Default: `all`
 - Options: `all` `authenticate` `authorize` `cache` or `proxy`
 
-Service mode sets the pomerium service(s) to run. If testing, you may want to set to `all` and run pomerium in "all-in-one mode." In production, you'll likely want to spin up several instances of each service mode for high availability.
+Service mode sets what service(s) to run. If testing, you may want to set to `all` and run pomerium in "all-in-one mode." In production, you'll likely want to spin up several instances of each service mode for high availability.
```
Comment (travisgroth) [inline docs/reference/readme.md:511]:
```suggestion
Service mode sets which service(s) to run. If testing, you may want to set to `all` and run pomerium in "all-in-one mode." In production, you'll likely want to spin up several instances of each service mode for high availability.
```

File: docs/reference/readme.md [hunk 4]
```diff
@@ -578,7 +579,7 @@ Many tracing backends support zipkin either directly or through intermediary age
 - Default: `/oauth2/callback`
 - Optional
 
-The authenticate callback path is the path/url from the authenticate service that will receive the response from your identity provider. The value must exactly match one of the authorized redirect URIs for the OAuth 2.0 client.
+The authenticate callback path is the path the authenticate service that will receive the callback response from your identity provider. The value must exactly match one of the authorized redirect URIs for the OAuth 2.0 client.
```
Comment (travisgroth) [inline docs/reference/readme.md:582]:
```suggestion
Authenticate callback path sets the path at which the authenticate service receives callback responses from your identity provider. The value must exactly match one of the authorized redirect URIs for the OAuth 2.0 client.
```

File: docs/reference/readme.md [hunk 5]
```diff
@@ -769,13 +791,13 @@ By default, conservative [secure HTTP headers](https://www.owasp.org/index.php/O
 - Example: `email`,`groups`, `user`
 - Optional
 
-The JWT Claim Headers setting allows you to pass specific user session data down to downstream applications as HTTP request headers. Note, unlike the header `x-pomerium-jwt-assertion` these values are not signed by the authorization service.
+The JWT Claim Headers setting allows you to pass specific user session data down to upstream applications as HTTP request headers. Note, unlike the header `x-pomerium-jwt-assertion` these values are not signed by the authorization service.
```
Comment (travisgroth) [inline docs/reference/readme.md:794]: 😆

File: docs/reference/readme.md (hunk 1 above)
Comment (desimone) [reply docs/reference/readme.md]: How about now?

File: docs/reference/readme.md [hunk 6]
```diff
@@ -18,11 +18,15 @@ Pomerium can be configured using a configuration file ([YAML]/[JSON]/[TOML]) or
 
 Using both [environmental variables] and config file keys is allowed and encouraged (for instance, secret keys are probably best set as environmental variables). However, if duplicate configuration keys are found, environment variables take precedence.
 
-Pomerium will automatically reload the configuration file if it is changed. At this time, only policy is re-configured when this reload occurs, but additional options may be added in the future. It is suggested that your policy is stored in a configuration file so that you can take advantage of this feature.
+:::tip
+
+Pomerium can hot-reload route configuration details, certificates, and other proxy settings.
```
Comment (travisgroth) [inline docs/reference/readme.md:23]:
```suggestion
Pomerium can hot-reload route configuration details, authorization policy, certificates, and other proxy settings.
```

**Pull request merged**